### PR TITLE
Fix identity

### PR DESCRIPTION
**Pull request by rafaelfolco**
Append identity variables to file when args are provided.
Include a test to cover this scenario.

### CODE REVIEW

**Comment by k-rister** [comment]
Why is the CI check failing?

**Comment by rafaelfolco** [comment]
> Why is the CI check failing?

fixed permissions for the identity file

File: tests/test-installer [hunk 1]
```diff
@@ -40,3 +40,16 @@ test "$?" = "0" || exit 1
...
+touch /tmp/auth-file.json || true
+cfgfile=$(grep SYSCONFIG= crucible-install.sh | cut -d '=' -f2 | sed 's/"//g')
+idfile=$(grep IDENTITY= crucible-install.sh | cut -d '=' -f2 | sed 's/"//g')
+# TODO(rfolco): temporary workaround until we make it distro generic
+sudo mkdir -p $(dirname $cfgfile)
+sudo ./crucible-install.sh \
+  --registry myregistry.io/crucible \
+  --auth-file /tmp/auth-file.json \
+  --name "Nobody" \
+  --email "nobody@somewhere.com"
```
Comment (k-rister) [inline tests/test-installer:54]: Unless I'm not understanding things I don't think this worked in the CI test.  It says the test passed, but I don't see these values in the following grep output:

https://github.com/perftool-incubator/crucible/pull/140/checks#step:3:128

**Comment by rafaelfolco** [comment]
@k-rister  I had to change the logic around identity a little. Please see if this is reasonable.

**Review by k-rister — APPROVED** [review]

**Comment by k-rister** [comment]
Can you briefly describe what you had to change?  Seems like it would good understand that going forward...

**Comment by rafaelfolco** [comment]
> Can you briefly describe what you had to change? Seems like it would good understand that going forward...

Summary:
Logic around identity has become a little complex because we may end up with many scenarios there. 

The problem:
Identity was overriding args specified from cli and the test was failing.

Fix:
I changed the precedence of loading name and email config. If --name or --email are specified, then skip loading identity file. Then check if name and email are specified (no matter if it was from args or from file) and read as input otherwise.
At the end, override changes in the identity file.

**Pull request merged**